src/client/client.ts

```diff
@@ -67,32 +68,66 @@ if (!Symbol.asyncIterator) {
  * @public
  */
 export default class TaskClient {
+  private _client: CosmosDbClient;
+  private _options: Required<TaskClientOptions>;
+  private _interceptor: InterceptorProcessor;
+
+  /**
+   * Create a client using a pre-initialized Azure Cosmos DB client. Usage of
+   * this API is only recommended for advanced users. Most users should use
+   * the TaskClient.create method instead.
+   *
+   * @param cosmosClient    - Client created from a {@link
+   *                          @azure/cosmos#Container} that is scoped to the
+   *                          collection/container that the tasks client
+   *                          should use.
+   * @param options         - Client creation options
+   *
+   * @internal
+   */
+  private constructor(
+    cosmosClient: CosmosDbClient,
+    options?: TaskClientOptions
+  ) {
+    this._client = cosmosClient;
+    this._options = {
+      lockDurationMs: 30000,
+      maxExecutionTimeMs: 30 * 60 * 1000,
+      pollIntervalMs: 5000,
+      maxUpdateParallelism: 10,
+      interceptors: {},
+      retries: INTERNAL_RETRY_OPTIONS,
+      ...options
+    };
+    this._interceptor = new InterceptorProcessor(this._options.interceptors);
+  }
+
   /**
    * Initializes a task client using the provided account information,
    * creating the specified database and collection if necessary.
    *
-   * @param account     - Azure Cosmos DB account url
-   * @param database    - Azure Cosmos DB database name
-   * @param collection  - Azure Cosmos DB collection/container name
-   * @param key         - Azure Cosmos DB account key
-   * @param options     - Client creation options
+   * @param account         - Azure Cosmos DB account url
+   * @param database        - Azure Cosmos DB database name
+   * @param collection      - Azure Cosmos DB collection/container name
+   * @param aadCredentials  - Azure Identity Library ChainedTokenCredential
+   * @param options         - Client creation options
    *
    * @returns Promise containing the initialized client
    *
    * @public
    */
-  static async create(
+  static async createFromCredential(
     account: string,
     database: string,
     collection: string,
-    key: string,
+    aadCredentials: ChainedTokenCredential,
```

Comment:
The use of `ChainedTokenCredential` still seems too specific here. I'm assuming the cosmos SDK just takes a `TokenCredential`? That should be sufficient for this scenario

```suggestion
    aadCredentials: TokenCredential,
```